File: Sample/module1/src/main/java/com/chenenyu/router/module/Module1Activity.java
package com.chenenyu.router.module;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.chenenyu.router.Router;
import com.chenenyu.router.annotation.Route;
import com.chenenyu.router.module1.R;

@Route("module1")
public class Module1Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_module1);

        Fragment fragment = (Fragment) Router.build("fragment2").getFragment(this);
        if (fragment != null) {
            getSupportFragmentManager().beginTransaction().add(R.id.activity_module1, fragment).commit();
        }
    }

}

File: Sample/module1/src/main/kotlin/com/chenenyu/router/module/Module1Activity.kt
package com.chenenyu.router.module

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.chenenyu.router.Router
import com.chenenyu.router.annotation.Route
import com.chenenyu.router.module1.databinding.ActivityModule1Binding

@Route("module1", "router://filter/module1")
class Module1Activity : AppCompatActivity() {

    private lateinit var binding: ActivityModule1Binding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityModule1Binding.inflate(layoutInflater)
        setContentView(binding.root)
        val fragment = Router.build("fragment2").getFragment(this) as Fragment
        supportFragmentManager.beginTransaction().add(binding.activityModule1.id, fragment).commit()
    }
}
